extension/src/experiments/columns/model.ts

```diff
@@ -22,6 +25,9 @@ export class ColumnsModel extends PathSelectionModel<Column> {
       PersistenceKey.METRICS_AND_PARAMS_COLUMN_WIDTHS,
       {}
     )
+    this.columnHeadLayersMaxLength =
```

Comment:
Decided to store the config value in the Model vs in a lower level function so that the config is only collected once (vs every time a column is created for example), but this means that the variable is passed over several different functions to get to `limitAncestorDepth`.

Reply:
`getConfigValue(ConfigKey.EXP_TABLE_HEAD_MAX_LAYERS)` is not an expensive call. I don't think we need to store the value. Making the call also means that we don't need to add a listener.

See https://github.com/iterative/vscode-dvc/blob/998d3f3496ac451c7c32f1a59be4477286b0f68e/extension/src/test/suite/util.ts#L47-L48 for an example of a configuration change event listener.

(also fixes [this](https://github.com/iterative/vscode-dvc/pull/2436/files#r977045140)).

Reply:
Will still need a listener to fire the tree/table updated event emitter(s) whenever the config changes. See https://github.com/iterative/vscode-dvc/blob/bb10bcce4cd85e30e3b53aa0825e9b3273625d98/extension/src/experiments/workspace.ts#L24

Reply:
Thanks for the info!